core/packages/PackageDB.cc

```diff
@@ -165,6 +166,31 @@ const PackageInfo &PackageDB::getPackageForFile(const core::GlobalState &gs, cor
     return NONE_PKG;
 }
 
+const PackageInfo &PackageDB::getPackageInfo(const core::GlobalState &gs, std::string_view nameStr) const {
+    auto mangled = absl::StrCat(absl::StrReplaceAll(nameStr, {{"::", "_"}}), core::PACKAGE_SUFFIX);
+    auto utf8Name = gs.lookupNameUTF8(mangled);
+    if (!utf8Name.exists()) {
+        return NONE_PKG;
+    }
+
+    auto packagerName = gs.lookupNameUnique(core::UniqueNameKind::Packager, utf8Name, 1);
+    if (!packagerName.exists()) {
+        return NONE_PKG;
+    }
+
+    auto cnst = gs.lookupNameConstant(packagerName);
+    if (!cnst.exists()) {
+        return NONE_PKG;
+    }
+
+    auto it = this->packages_.find(cnst);
+    if (it == this->packages_.end()) {
+        return NONE_PKG;
+    }
+
+    return *it->second;
```

Comment:
```suggestion
    return getPackageInfo(cnst);
```
Minor, but I believe this could be simplified.

Reply:
Oh hey, great point!